internal/pally/registry.go

```diff
@@ -0,0 +1,250 @@
+package pally
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/prometheus/common/model"
+	"github.com/uber-go/tally"
+	"go.uber.org/atomic"
+)
+
+// A Registry is a collection of metrics, usually scoped to a single package or
+// object. Each Registry is also its own http.Handler, and can serve
+// Prometheus-flavored text and protocol buffer pages for metrics introspection
+// or scraping.
+type Registry struct {
+	http.Handler
```

Comment:
Is making the Handler public intentional? Do we want people to be able to do, 

    r.Handler = myHandler